sui/publish-gateway-v2.js

```diff
@@ -0,0 +1,183 @@
+const { SuiClient, getFullnodeUrl } = require('@mysten/sui.js/client');
+const { Ed25519Keypair } = require('@mysten/sui.js/keypairs/ed25519');
+const secp256k1 = require('secp256k1');
+const { saveConfig, loadConfig } = require('../evm/utils');
+const { publishAll } = require('@axelar-network/axelar-cgp-sui/scripts/publish-all');
+const { getConfig, parseEnv } = require('@axelar-network/axelar-cgp-sui/scripts/utils');
+const { setTrustedAddresses } = require('@axelar-network/axelar-cgp-sui/scripts/its/set-trusted-address');
+const { setItsDiscovery } = require('@axelar-network/axelar-cgp-sui/scripts/its/discovery');
+const { transferOperatorship } = require('@axelar-network/axelar-cgp-sui/scripts/gateway');
+const { Command, Option } = require('commander');
+const { publishPackage } = require('@axelar-network/axelar-cgp-sui/scripts/publish-package');
+const { TransactionBlock } = require('@mysten/sui.js/transactions');
+const { bcs } = require('@mysten/sui.js/bcs');
+const { ethers } = require('hardhat');
+const {
+    Contract,
+    Wallet,
+    providers: { JsonRpcProvider },
+    utils: { parseEther, keccak256, defaultAbiCoder, arrayify, hexlify, randomBytes },
+    constants: { AddressZero },
+} = ethers;
+
+const { requestSuiFromFaucetV0 } = require('@mysten/sui.js/faucet');
+
+async function main(options) {
+    options.validatorAddresses = JSON.parse(options.validatorAddresses);
+    options.weights = JSON.parse(options.weights);
+    options.validators = options.validatorAddresses.map((val, index) => {return {signer: arrayify(val), weight: options.weights[index]}});
+    options.threshold = JSON.parse(options.threshold);
+    const privKey = Buffer.from(options.privateKey, 'hex');
+    const keypair = Ed25519Keypair.fromSecretKey(privKey);
+    const client = new SuiClient({ url: getFullnodeUrl(options.env) });
+    if (options.faucetUrl) {
```

Comment:
We should avoid doing too many things in one script, instead of creating dedicated scripts for separate tasks. Faucet url should be added to the configs, no need for an option.